app/src/main/java/com/github/bgabriel998/softwaredevproject/points/UserPoint.java

```diff
@@ -29,24 +35,33 @@
     private boolean customLocation;
 
     /**
-     * Constructor for the UserPoint.
+     * Constructor for the UserPoint. Private because it is a singleton.
      *
      * @param mContext  the current context of the app
      */
-    public UserPoint(Context mContext) {
+    private UserPoint(Context mContext) {
         super(GPSTracker.DEFAULT_LAT, GPSTracker.DEFAULT_LON, GPSTracker.DEFAULT_ALT);
         gpsTracker = new GPSTracker(mContext, this);
         customLocation = false;
     }
```

Comment:
```single_instance``` should be assigned here, else ```single_instance``` is always ```null```
```suggestion
        single_instance = this;
    }
```